cluster-autoscaler/processors/status/scale_down_status_processor.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator"
+)
+
+// ScaleDownStatus represents the state of scale down.
+type ScaleDownStatus struct {
+	Result            ScaleDownResult
+	ScaledDownNodes   []*ScaleDownNode
+	NodeDeleteResults map[string]error
```

Comment:
Given that nil is a meaningful value for success and a default value for non-existent key at the same time, I wonder if this is what we want to have here long-term. @MaciekPytel WDYT?

Reply:
Discussed with @towca, I think it's reasonable to explicitly check for whether key is in map or not.